sdk/python/lib/pulumi/runtime/stack.py

```diff
@@ -50,28 +60,32 @@ async def run_pulumi_func(func: Callable):
             log.debug(f"waiting for quiescence; {rpcs_remaining} RPCs outstanding")
             await RPC_MANAGER.rpcs.pop()
 
-        # Asyncio event loops require that all outstanding tasks be completed by the time that the
-        # event loop closes. If we're at this point and there are no outstanding RPCs, we should
-        # just cancel all outstanding tasks.
-        #
-        # We will occasionally start tasks deliberately that we know will never complete. We must
-        # cancel them before shutting down the event loop.
```

Comment:
Is this actually true? I haven't found any instances where we start tasks "forever" in practice. The solution here depends on this statement being false and an assumption that all futures will resolve unless an exception is thrown.